### PR TITLE
Implement PlantUmlParser

### LINKED ISSUE #8: Implement PlantUmlParser
## Overview
Parse PlantUML component/object diagrams into a `DependencyGraph` model.

## Tasks
- [ ] Define PlantUML grammar subset to support (components, dependencies, notes)
- [ ] Implement `PlantUmlParser.cs` in `Bartleby.Infrastructure/Graph/`
- [ ] Parse nodes (components/objects) from PlantUML
- [ ] Parse dependency arrows (`-->`, `..>`, etc.)
- [ ] Build `DependencyGraph` model from parsed data
- [ ] Handle parse errors gracefully

## Testing Requirements
- [ ] Unit tests for parsing valid PlantUML
- [ ] Unit tests for handling malformed PlantUML
- [ ] Unit tests for edge cases (empty files, comments, nested structures)
- [ ] Integration test with real `.puml` files

## Acceptance Criteria
- [ ] Parser extracts nodes and edges from PlantUML syntax
- [ ] Parser returns meaningful errors for invalid input
- [ ] All tests pass

## Example PlantUML to Support
```plantuml
@startuml
component "Task A" as A
component "Task B" as B
component "Task C" as C

A --> B : depends on
B --> C : depends on
@enduml
```

---
**Story**: Testable unit of code | **Parent Epic**: #2 PlantUML & Dependency Resolution

### PR DESCRIPTION
**Pull request by elephantgerald**
## Summary
- Implements `PlantUmlParser` with regex-based parsing for a PlantUML subset
- Supports node types: `component`, `object`, `rectangle`, `node`, `package`
- Supports arrows: `-->` (solid), `..>` (dashed), `==>` (bold), and reverse directions
- Parses labels, comments, and `@startuml`/`@enduml` blocks
- Adds `PlantUmlParseResult` models for structured parser output
- Integrates parser into `PlantUmlGraphStore.LoadGraphAsync()`
- Adds 35 unit tests covering valid PlantUML, malformed input, and edge cases

## Test plan
- [x] Unit tests for parsing valid PlantUML (multiple node types, arrows, labels)
- [x] Unit tests for handling malformed PlantUML (duplicate aliases, unknown nodes, unclosed blocks)
- [x] Unit tests for edge cases (empty files, comments only, mixed line endings)
- [x] All 38 tests pass (`dotnet test`)

## Future enhancement
Consider replacing regex-based parser with ANTLR + community PlantUML grammar for full syntax support.

Closes #8

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Bartleby.Infrastructure/Graph/PlantUmlParser.cs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    [GeneratedRegex(
+        @"^\s*(component|object|rectangle|node|package)\s+(\w+)\s*$",
+        RegexOptions.IgnoreCase)]
+    private static partial Regex NodeWithAliasOnlyRegex();
+
+    // Edge patterns: A --> B : label  OR  A --> B
+    // Supports: -->, ..>, ==>, <--, <.., <==
+    [GeneratedRegex(
+        @"^\s*(\w+)\s*(--|\.\.|\=\=)(\>)\s*(\w+)(?:\s*:\s*(.+))?$")]
+    private static partial Regex ForwardEdgeRegex();
```
Comment (elephantgerald) [inline src/Bartleby.Infrastructure/Graph/PlantUmlParser.cs]: Arrow parsing is a little too rigid, apparently PlantUML allows for variants
A --> B
A -> B
A -->o B
A o--> B
A ..> B
A ==> B
